pkg/controller/registry/reconciler/reconciler_test.go

```diff
@@ -167,7 +167,183 @@ func TestPodMemoryTarget(t *testing.T) {
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "name",
+							Image: "image",
+							Ports: []corev1.ContainerPort{{Name: "grpc", ContainerPort: 50051}},
+							ReadinessProbe: &corev1.Probe{
+								ProbeHandler: corev1.ProbeHandler{
+									Exec: &corev1.ExecAction{
+										Command: []string{"grpc_health_probe", "-addr=:50051"},
+									},
+								},
+								InitialDelaySeconds: 0,
+								TimeoutSeconds:      5,
+							},
+							LivenessProbe: &corev1.Probe{
+								ProbeHandler: corev1.ProbeHandler{
+									Exec: &corev1.ExecAction{
+										Command: []string{"grpc_health_probe", "-addr=:50051"},
+									},
+								},
+								InitialDelaySeconds: 0,
+								TimeoutSeconds:      5,
+							},
+							StartupProbe: &corev1.Probe{
+								ProbeHandler: corev1.ProbeHandler{
+									Exec: &corev1.ExecAction{
+										Command: []string{"grpc_health_probe", "-addr=:50051"},
+									},
+								},
+								FailureThreshold: 10,
+								PeriodSeconds:    10,
+								TimeoutSeconds:   5,
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("10m"),
+									corev1.ResourceMemory: resource.MustParse("50Mi"),
+								},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								ReadOnlyRootFilesystem: pointer.Bool(false),
+							},
+							ImagePullPolicy:          image.InferImagePullPolicy("image"),
+							TerminationMessagePolicy: "FallbackToLogsOnError",
+						},
+					},
+					NodeSelector:       map[string]string{"kubernetes.io/os": "linux"},
+					ServiceAccountName: "service-account",
+				},
+			},
+		},
+		{
+			name: "content extraction expected",
+			input: &v1alpha1.CatalogSource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testns",
+				},
+				Spec: v1alpha1.CatalogSourceSpec{
+					GrpcPodConfig: &v1alpha1.GrpcPodConfig{
+						ExtractContent: &v1alpha1.ExtractContentConfig{
+							CacheDir:   "/tmp/cache",
+							CatalogDir: "/catalog",
+						},
+					},
+				},
+			},
+			expected: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "test-",
+					Namespace:    "testns",
+					Labels:       map[string]string{"olm.pod-spec-hash": "7cc8fc96f8", "olm.managed": "true"},
+					Annotations:  map[string]string{"cluster-autoscaler.kubernetes.io/safe-to-evict": "true"},
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name:         "catalog-content",
+							VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}},
+						},
+					},
+					InitContainers: []corev1.Container{
+						{
+							Name:         "extract-content",
+							Image:        "image",
+							Command:      []string{"/bin/sh", "-c"},
+							Args:         []string{"cp /catalog /catalog/catalog && cp /tmp/cache /catalog/cache"},
```

Comment:
How do we handle this with custom catalogs?  For e.g. RH catalogs all expose the FBC catalog in `/configs`, looking like
```tree
configs
├── abinitio-runtime-operator
│   └── catalog.json
├── acc-operator
│   └── catalog.json
...
```

which is expressed in the entrypoint+cmd of the image, e.g. 
```bash
            ...
            "Cmd": [
                "serve",
                "/configs",
                "--cache-dir=/tmp/cache"
            ],
            ...
            "Entrypoint": [
                "/bin/opm"
            ],
            ...
```

but it is very much not a requirement to use these specific paths.  If someone did something like 
`opm serve /var/spool/cats --cache-dir=/var/cache/catalog`
would it work?

Reply:
The directories are part of the config in the CatalogSource, so the user can pass in whatever they want

Reply:
Duh!  I thought this was a filesystem ref, and missed where the init container would set this as a destination for the copy (with source from catsrc config).  
So if you're just consistently `cp -r[p]` everwhere, then you're good regardless.